nixos/tests/jellyfin.nix

```
@@ -1,16 +1,151 @@
-import ./make-test-python.nix ({ lib, ...}:
-
-{
-  name = "jellyfin";
-  meta.maintainers = with lib.maintainers; [ minijackson ];
-
-  machine =
-    { ... }:
-    { services.jellyfin.enable = true; };
-
-  testScript = ''
-    machine.wait_for_unit("jellyfin.service")
-    machine.wait_for_open_port(8096)
-    machine.succeed("curl --fail http://localhost:8096/")
-  '';
-})
+import ./make-test-python.nix ({ lib, pkgs, ... }:
+
+  {
+    name = "jellyfin";
+    meta.maintainers = with lib.maintainers; [ minijackson ];
+
+    machine =
+      { ... }:
+      {
+        services.jellyfin.enable = true;
+        environment.systemPackages = with pkgs; [ ffmpeg ];
+      };
+
+    # Documentation of the Jellyfin API: https://api.jellyfin.org/
+    # Beware, this link can be resource intensive
+    testScript =
+      let
+        dummyAuthHeader = ''
+          MediaBrowser Client="NixOS Integration Tests", DeviceId="1337", Device="Apple II", Version="20.09"'';
+
+        # These use Python string interpolation, remember to use f""
+        get = path: "curl --fail 'http://localhost:8096${path}' -H 'X-Emby-Authorization:{auth_header}'";
+        post = path:
+          "curl --fail -X POST 'http://localhost:8096${path}' -d '' -H 'X-Emby-Authorization:{auth_header}'";
+        postWith = path: json:
+          "curl --fail -X POST 'http://localhost:8096${path}' -d '@${json}' -H Content-Type:application/json -H 'X-Emby-Authorization:{auth_header}'";
+
+        payloads = {
+          auth = pkgs.writeText "auth.json" (builtins.toJSON {
+            Username = "jellyfin";
+          });
+          empty = pkgs.writeText "empty.json" (builtins.toJSON { });
+        };
+      in
+      ''
+        import json
+        import time
+        from urllib.parse import urlencode
+
+        machine.wait_for_unit("jellyfin.service")
+        machine.wait_for_open_port(8096)
+        machine.succeed("curl --fail http://localhost:8096/")
+
+        time.sleep(1)
+
+        health = machine.succeed("curl --fail http://localhost:8096/health")
```

Maybe

```suggestion
        machine.wait_until_succeeds("curl --fail http://localhost:8096/health | grep Healthy")
```

instead - and drop the time.sleep above.